File: concepts/data-structures/linked-lists/linkedList/linkedList.cpp
#include <iostream>
using namespace std;

template <class T>
class Node {
 public:
  T data;
  Node<T> *next = NULL;
  Node<T>(T data) : data(data){};
};

template <class T>
class LinkedList {
 private:
  Node<T> *_head = NULL;

 public:
  void append(T data) {
    if (_head == NULL) {
      _head = new Node<T>(data);
      return;
    }
    Node<T> *current = _head;
    while (current->next != NULL) {
      current = current->next;
    }
    current->next = new Node<T>(data);
  };

  void deleteIndex(int index) {
    Node<T> *current = _head;
    for (int i=0;i<index-1;++i) {
      current = current->next;
    }
    Node<T> *toDelete = current->next;
    current->next = current->next->next;
    delete toDelete;
  }

  void print() {
    Node<T> *current = _head;
    while (current != NULL) {
      cout << current->data << " ";
      current = current->next;
    }
    cout << "\b\n";
  }
};

int main() {
  LinkedList<int> llist{};
  llist.append(10);
  llist.append(21);
  llist.append(69);
  llist.print();
  llist.deleteIndex(1);
  llist.print();
  return 0;
}

File: concepts/data-structures/linked-lists/linkedList/linkedList.py
from __future__ import annotations

class Node:
  data: str
  next: Node = None
  def __init__(self, data: str):
    self.data = data

class LinkedList:
  head: Node = None
  
  def append(self, data: str):
    if self.head == None:
      self.head = Node(data)
      return
    cur = self.head
    while cur.next != None:
      cur = cur.next
    cur.next = Node(data)

  def removeAtIndex(self, index: int):
    n = 0
    cur = self.head
    while cur.next != None and n < index:
      n += 1
      cur = cur.next
    cur = cur.next.next
    # del cur.next

  def print(self):
    arr = []
    cur = self.head
    while cur != None:
      arr.append(cur.data)
      cur = cur.next
    print(arr)

if __name__ == '__main__':
  llist = LinkedList()
  llist.append("1")
  llist.append("2")
  llist.append("3")
  llist.append("4")
  llist.append("5")
  llist.append("6")
  llist.removeAtIndex(1)
  llist.print()